### PR TITLE
Improved error handling

### PR DESCRIPTION
**Pull request by mmiedlarz**

### CODE REVIEW

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    public static let DefaultCode:   String = "creatubbles-apiclient-default-code"
+    public static let DefaultTitle:  String = "creatubbles-apiclient-default-title"
+    public static let DefaultSource: String = "creatubbles-apiclient-default-source"
+    public static let DefaultDetail: String = "creatubbles-apiclient-default-detail"
+}
+
+//MARK: Error codes
+extension APIClientError
+{
+    static let UnknownStatus: Int = -6001
```
Comment (mmiedlarz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:43]: Change to public

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    static let LoginStatus: Int = -6002
+    static let UploadCancelledStatus: Int = -6003
+}
+
+public class APIClientError: ErrorType
```
Comment (mmiedlarz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:48]: Add URL for APIClientError documentation:
https://partners.creatubbles.com/api/#errors
http://jsonapi.org/format/#error-objects

File: CreatubblesAPIClient/Sources/Core/Error/ErrorTransformer.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+class ErrorTransformer
+{
+    static func errorFromResponse(response: Dictionary<String, AnyObject>?, error: ErrorType?) -> APIClientError?
+    {
+        if let err = error as? APIClientError
+        {
+            return err
+        }
+        if let err = errorsFromResponse(response).first
```
Comment (mmiedlarz) [inline CreatubblesAPIClient/Sources/Core/Error/ErrorTransformer.swift:38]: To investigate - how can we support many errors in current APIClient implementation?

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+}
+
+extension APIClientError
+{
+    class var genericLoginError: APIClientError
+    {
+        return APIClientError(status: APIClientError.LoginStatus,
+                              code:   "authentication-error",
+                              title:  "Authentication failed",
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:84]: Shouldn't it be localized?

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                              source: "https://www.creatubbles.com/api/v2/users",
+                              detail: "Authentication failed. No more details are available at the moment. Please try again.")
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:86]: Localization?

Comment (pleszkiewicz) [reply CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:86]: Is it user-friendly to show the same text in both title and body? 

```
         Authentication failed
Authentication failed. No more...
```

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    
+    class var genericUploadCancelledError: APIClientError
+    {
+        return APIClientError(status: APIClientError.UploadCancelledStatus,
+                              code:   "upload-cancelled",
+                              title:  "Upload cancelled",
+                              source: APIClientError.DefaultSource,
+                              detail: "Upload cancelled")
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:95]:
```
"Upload cancelled" 
"Upload cancelled"
```

Missing localizations.

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        self.domain = APIClientError.DefaultDomain
+    }
+}
+
+extension APIClientError
+{
+    class var genericLoginError: APIClientError
+    {
+        return APIClientError(status: APIClientError.LoginStatus,
+                              code:   "authentication-error",
```
Comment (pleszkiewicz) [inline CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:83]: Can you store this value in any variable? It is reused in other place.

**Review by pleszkiewicz — CHANGES_REQUESTED** [review]
Code: +1
Func: n/a

File: CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift (hunk 6 above)
Comment (mmiedlarz) [reply CreatubblesAPIClient/Sources/Core/Error/APIClientError.swift:95]: right now we don't support any localizations in APIClient

**Pull request merged**